internal/testhelpers/e2e/session.go

```diff
@@ -220,76 +212,125 @@ func NewNoPathUpdate(t *testing.T, retainDirs bool, extraEnv ...string) *Session
+func (s *Session) Spawn(args ...string) *SpawnedCmd {
+	return s.SpawnCmdWithOpts(s.Exe, OptArgs(args...))
 }
 
 // SpawnWithOpts spawns the state tool executable to be tested with arguments
-func (s *Session) SpawnWithOpts(opts ...SpawnOptions) *termtest.ConsoleProcess {
+func (s *Session) SpawnWithOpts(opts ...SpawnOptSetter) *SpawnedCmd {
 	return s.SpawnCmdWithOpts(s.Exe, opts...)
 }
 
 // SpawnCmd executes an executable in a pseudo-terminal for integration tests
-func (s *Session) SpawnCmd(cmdName string, args ...string) *termtest.ConsoleProcess {
-	return s.SpawnCmdWithOpts(cmdName, WithArgs(args...))
+func (s *Session) SpawnCmd(cmdName string, args ...string) *SpawnedCmd {
+	return s.SpawnCmdWithOpts(cmdName, OptArgs(args...))
 }
 
 // SpawnShellWithOpts spawns the given shell and options in interactive mode.
-func (s *Session) SpawnShellWithOpts(shell Shell, opts ...SpawnOptions) *termtest.ConsoleProcess {
+func (s *Session) SpawnShellWithOpts(shell Shell, opts ...SpawnOptSetter) *SpawnedCmd {
 	if shell != Cmd {
-		opts = append(opts, AppendEnv("SHELL="+string(shell)))
+		opts = append(opts, OptAppendEnv("SHELL="+string(shell)))
 	}
+	opts = append(opts, OptRunInsideShell(false))
 	return s.SpawnCmdWithOpts(string(shell), opts...)
 }
 
 // SpawnCmdWithOpts executes an executable in a pseudo-terminal for integration tests
-// Arguments and other parameters can be specified by specifying SpawnOptions
-func (s *Session) SpawnCmdWithOpts(exe string, opts ...SpawnOptions) *termtest.ConsoleProcess {
-	if s.cp != nil {
-		s.cp.Close()
-	}
+// Arguments and other parameters can be specified by specifying SpawnOptSetter
+func (s *Session) SpawnCmdWithOpts(exe string, optSetters ...SpawnOptSetter) *SpawnedCmd {
+	spawnOpts := NewSpawnOpts()
+	spawnOpts.Env = s.Env
+	spawnOpts.Dir = s.Dirs.Work
+
+	spawnOpts.TermtestOpts = append(spawnOpts.TermtestOpts,
+		termtest.OptErrorHandler(func(tt *termtest.TermTest, err error) error {
+			s.t.Fatal(s.DebugMessage(errs.JoinMessage(err)))
+			return err
+		}),
+		termtest.OptDefaultTimeout(defaultTimeout),
+		termtest.OptCols(140),
+		termtest.OptRows(30), // Needs to be able to accommodate most JSON output
+	)
 
-	env := s.Env
+	// Work around issue where multiline values sometimes have the wrong line endings
+	// See for example TestBranch_List
+	// https://activestatef.atlassian.net/browse/DX-2169
+	spawnOpts.TermtestOpts = append(spawnOpts.TermtestOpts,
+		termtest.OptNormalizedLineEnds(true),
+	)
 
-	pOpts := Options{
-		Options: termtest.Options{
-			DefaultTimeout: defaultTimeout,
-			Environment:    env,
-			WorkDirectory:  s.Dirs.Work,
-			RetainWorkDir:  true,
-			ObserveExpect:  observeExpectFn(s),
-			ObserveSend:    observeSendFn(s),
-		},
-		NonWriteableBinDir: false,
+	for _, optSet := range optSetters {
+		optSet(&spawnOpts)
+	}
+
+	var shell string
+	var args []string
+	if spawnOpts.RunInsideShell {
+		switch runtime.GOOS {
+		case "windows":
+			shell = "cmd.exe"
```

Comment:
I see you added the `Cmd` constant on line 295, but you didn't change the one here. Unless I'm missing something, best to be safe and change it here too, right?